core/eolearn/core/eotask.py

```diff
@@ -68,19 +68,19 @@ def _execute_handling(self, *eopatches, **kwargs):
         """
         self.private_task_config.start_time = datetime.datetime.now()
 
-        caught_exception = None
         try:
             return_value = self.execute(*eopatches, **kwargs)
+            self.private_task_config.end_time = datetime.datetime.now()
+            return return_value
         except BaseException as exception:
-            caught_exception = exception, sys.exc_info()[2]
-
-        if caught_exception is not None:  # Exception is not raised in except statement to prevent duplicated traceback
-            exception, traceback = caught_exception
-            raise type(exception)('During execution of task {}: {}'.format(self.__class__.__name__,
-                                                                           exception)).with_traceback(traceback)
-
-        self.private_task_config.end_time = datetime.datetime.now()
-        return return_value
+            exception, traceback = exception, sys.exc_info()[2]
+
+            # Some special exceptions don't accept error message as a parameter and a TypeError is raised in such case.
+            try:
+                errmsg = 'During execution of task {}: {}'.format(self.__class__.__name__, exception)
+                raise type(exception)(errmsg).with_traceback(traceback)
```

Comment:
There is an edge case if `exception` is already a `TypeError`. In such case this wouldn't produce an expected result. I suggest a slight restructuring:
```Python
try:
    errmsg = 'During execution of task {}: {}'.format(self.__class__.__name__, exception)
    improved_exception = type(exception)(errmsg)
except TypeError:
    improved_exception = exception

raise improved_exception.with_traceback(traceback)
```
This way the internal try-catch will only catch `TypeError` obtained from exception initialization and not from raising the final exception.

Reply:
Good one, I agree.